lib/CXGN/BrAPI/v2/ObservationVariables.pm

```diff
@@ -174,6 +170,31 @@ sub search {
         }
     }
 
+    if (scalar(@study_ids)>0){
+        my $trait_ids_sql;
+
+        foreach my $study_id (@study_ids){
+            my $study_check = $self->bcs_schema->resultset('Project::Project')->find({project_id=>$study_id});
+            if ($study_check) {
+                my $t = CXGN::Trial->new({ bcs_schema => $self->bcs_schema, trial_id => $study_id });
+                my $traits_assayed = $t->get_traits_assayed();
+                
+                foreach (@$traits_assayed){
+                    $trait_ids_sql .= ',' . $_->[0] ;
+                }
+            }
+        }
+
+        $trait_ids_sql =~ s/^,//g;
+
+        if ($trait_ids_sql){
+            push @and_wheres, "cvterm.cvterm_id IN ($trait_ids_sql)";
+        } else {
+            return CXGN::BrAPI::JSONResponse->return_error($self->status, sprintf('Variables not found for the searched studyDbId'), 400);
```

Comment:
I think this was probably pulled in from Mirella but should this just return an empty response rather than error since it's a query parameter or search? Guess we'll have to see how Field Book handles that

Reply:
I think since we haven't tested FieldBook against Breedbase brapi v2 that this can part of that work.